Resources/Private/Javascript/Components/AddToCart.js

```diff
@@ -7,12 +7,15 @@ const addToCartOnClick = ({ proxy, productId }) => {
     event.preventDefault();
     const result = await apiClient.addProductToCart(productId);
 
-    document.dispatchEvent(new Event('cart-changed'));
-    document.dispatchEvent(new CustomEvent('added-to-cart', {
-      detail: {
-        message: result
-      }
-    }));
+    if(result.status == 200) {
+      document.dispatchEvent(new Event('cart-changed'));
+      document.dispatchEvent(new CustomEvent('added-to-cart', {
```

Comment:
it did not add something to the cart, but you should have an event, when the action is finished, you might want to react to a 4xx or 5xx as well without using the error event e.g. by removing a loading state